compiler/rustc_builtin_macros/src/test.rs

```diff
@@ -80,7 +114,17 @@ pub fn expand_test_or_bench(
     }
 
     let (item, is_stmt) = match item {
-        Annotatable::Item(i) => (i, false),
+        Annotatable::Item(i) => {
+            if let Some(attr) = i.attrs.iter().filter(|attr| is_test_attribute(attr)).next() {
+                let sd = &cx.sess.parse_sess.span_diagnostic;
+                sd.struct_span_warn(
```

Comment:
Hard-coded warnings should almost never be used, either lints or errors are supposed to be used instead.
I need to think what is a better course of action in this specific case.

Reply:
We cannot recognize other calls to `#[test]` precisely here due to cases like this
```rust
#[test] // built-in `test`
#[arbitrary_attribute]
#[test] // in theory this can resolve to a different non-builtin macro attribute
        // or this can be a significant input to `#[arbitrary_attribute]`
```
, so if we error on this it should be a documented part of the `test`'s behavior.

Reply:
TBH, doing nothing and closing #90979 as not-an-issue would also be a good solution.

Reply:
I suggest adding a new warn-by-default lint called `duplicate_macro_attributes`, and reuse the wording from https://github.com/rust-lang/rust/pull/88681.